C_CPP/Algorithmic Approaches/Sorting/MergeSort.MD

```diff
@@ -0,0 +1,84 @@
+# MERGE SORT
+- MERGE SORT: Merge sort is one of the widely accepted sorting algorithm. It is based on Divide and Conquer algorithm. In this type of algorithm we build a heap and sort with comparisons.</br>
+
+-  Merge Sort is accomplished by Dividing the array into two halves, and then again for the two halves until we get unit, and finally merging the two sorted halves. Merging is done ithe help of a Merge function
+
+## ALGORITHM
+```
+1. Call for Merge sort. \\ MergeSort(arr[],l,r)
+2. Find the middle point to divide array into two halves
+3. Call Merge sort for first half. \\ MergeSort(arr,l,m)
+4. Call Merge sort for second half. \\ MergeSort(arr,,mid+1,r)
+5. Merge the two sorted halves. \\ Merge(arr,l,m,r)
+```
+
+![download](https://user-images.githubusercontent.com/71391631/135991725-06b19089-c01b-4783-aaee-c26557a8c9d1.png)
```

Comment:
@rajraii Good going!
Try to enlarge/enhance this image a bit, so that it is more understandable.